service/history/queue/timer_queue_processor.go

```diff
@@ -425,12 +426,15 @@ func (t *timerQueueProcessor) completeTimer() error {
 	}
 
 	newAckLevelTimestamp := newAckLevel.(timerTaskKey).visibilityTimestamp
-	t.logger.Debugf("Start completing timer task from: %v, to %v", t.ackLevel, newAckLevelTimestamp)
 	if !t.ackLevel.Before(newAckLevelTimestamp) {
+		t.logger.Debugf("Skipping timer task completion because new ack level %v is not before ack level %v", newAckLevelTimestamp, t.ackLevel)
```

Comment:
nit: Consider dynamic log tags for easier log grouping. `tag.Dynamic` should work